viewing_party/party.py

```diff
@@ -31,19 +31,57 @@ def watch_movie(user_data, title):
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+# my own work with the help of tutor
+def get_watched_avg_rating(user_data):
+    count = 0
+    average_rating = 0
+
+    for i in range(len(user_data["watched"])):
+        count += (user_data["watched"][i]["rating"])
+        average_rating = count/len(user_data["watched"])
+    return average_rating
+
+def get_most_watched_genre(user_data):
+    genre_fave = [] # ['Fantasy', 'Fantasy', 'Fantasy', 'Action', 'Intrigue', 'Intrigue']
+    count = 0
+    current_highest = ""
+    
+    for users_list in user_data["watched"]:
+        genre_fave.append(users_list["genre"])
+        
+    for element in genre_fave:
+        if count < genre_fave.count(element): # count = 0 vs action=1
+            current_highest = element
+            count = genre_fave.count(element) # =1
+
+    if len(user_data["watched"]) == 0:
+        return None
+
+    return current_highest

```

Comment:
This works great! A few notes:
- Consider putting lines 57-58 at the beginning of the function. It's often a good idea to do our input validation at the beginning.
- I like the use of the for-each loop on line 49, but I think the variable name is a bit misleading. `user_list` is actually a dictionary representing a movie, not a user or a list.